### PR TITLE
Clean up alpaka::ignore_unused usages

### PR DESCRIPTION
**Pull request by BenjaminW3**

### CODE REVIEW

**Comment by ax3l** [comment]
oh no, we pushed the same cleanup ;)
#584

**Comment by ax3l** [comment]
We can merge the first and then you rebase this one? You catched a few more includes as it looks and this might work well.

**Comment by BenjaminW3** [comment]
@ax3l I have rebased my changes.

File: test/unit/rand/src/RandTest.cpp [hunk 1]
```diff
@@ -67,17 +66,19 @@ class RandTestKernel
             auto const r = dist(gen);
 #if !BOOST_ARCH_PTX
```
Comment (ax3l) [inline test/unit/rand/src/RandTest.cpp]: probably `#if BOOST_ARCH_PTX #else` would be easier to read?

File: test/unit/rand/src/RandTest.cpp [hunk 2]
```diff
@@ -67,17 +66,19 @@ class RandTestKernel
...
-#endif
+#else
             alpaka::ignore_unused(r);
+#endif
         }
 
         {
             auto dist(alpaka::rand::distribution::createNormalReal<double>(acc));
             auto const r = dist(gen);
 #if !BOOST_ARCH_PTX
```
Comment (ax3l) [inline test/unit/rand/src/RandTest.cpp]: probably `#if BOOST_ARCH_PTX #else` would be easier to read?

**Pull request merged**